EIPS/eip-2315.md

```diff
@@ -150,55 +307,11 @@ Bytecode: `0x6005565c5d5b60035e` (`PUSH1 0x05, JUMP, BEGINSUB, RETURNSUB, JUMPDE
 
 Consumed gas: `30`
 
-### Error on "walk-into-subroutine"
-
-In this example, the code 'walks' into a subroutine, which is not allowed, and causes an error
-
-Bytecode: `0x5c5d00` (`BEGINSUB, RETURNSUB, STOP`)
-
-
-|  Pc   |      Op     | Cost |   Stack   |   RStack  |
-|-------|-------------|------|-----------|-----------|
-|    0  |   BEGINSUB  |    2 |        [] |        [] |
-
-```
-Error: at pc=0, op=BEGINSUB: invalid subroutine entry
-```
-
-**Note 5**: The content of the error message, (`invalid subroutine entry`) is implementation-specific.
-
-## Implementations
-
-Three clients have implemented this (or an earlier version of) this proposal:
-
-- [geth](https://github.com/ethereum/go-ethereum/pull/20619) .
-- [besu](https://github.com/hyperledger/besu/pull/717), and
-- [openethereum](https://github.com/openethereum/openethereum/pull/11629).
-
-
-### Costs and Codes
-
-We suggest that the cost of 
-
-- `BEGINSUB` be _base_ (`2`)
-  - Although formally specified, the cost of `BEGINSUB` does not matter in practice, since `BEGINSUB` never executes without error. 
-- `JUMPSUB` be _high_ (`10`)
-  - This is the same as `JUMPI`, and `2` more than `JUMP`.
-- `RETURNSUB` be _low_ (`5`).
-
-Benchmarking might be needed to tell if the costs are well-balanced. 
-
-We suggest the following opcodes:
-
-```
-0x5c BEGINSUB
-0x5d RETURNSUB
-0x5e JUMPSUB
-```
-
-## Security Considerations
-
-These changes do introduce new flow control instructions, so any software which does static/dynamic analysis of evm-code
-needs to be modified accordingly. The `JUMPSUB` semantics are similar to `JUMP` (but jumping to a `BEGINSUB`), whereas the `RETURNSUB` instruction is different, since it can 'land' on any opcode (but the possible destinations can be statically inferred).
+## References
+Charles H.Moore, Geoffrey C. Leach. [FORTH - A Language for Interactive Computing](http://www.ultratechnology.com/4th_1970.pdf). 1970
+Philip J. Koopman, Jr. [Stack Computers: the new wave](https://users.ece.cmu.edu/~koopman/stack_computers/index.html). 1989
+Gavin Wood. [Ethereum:  A  Secure  Decentralized Generalized  Transaction  Ledger](https://ethereum.github.io/yellowpaper/paper.pdf). 2015-2021
+Martin Lundfall.  [EIP-2327: BEGINDATA Opcode](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-2327.md). 2019
+Nick Johnson. [EIP-3337: Frame pointer support for memory load and store operations](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-3337.md). 2021
 
-**Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).**
+**Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).**
```

Comment:
```suggestion
## Copyright
Copyright and related rights waived via [CC0](https://creativecommons.org/publicdomain/zero/1.0/).
```

Reply:
For some reason I can't just commit the suggestion, but have fixed my working copy.  Thanks.